knowledge-base/combobox-virtualization-loader.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Displaying Loading Indicator in ComboBox with Remote Data and Virtualization
+description: Learn how to add a loading indicator in the TelerikComboBox component when using remote data and virtualization functionality in UI for Blazor.
+type: how-to
+page_title: Adding Loader to ComboBox During Remote Data Fetch and Virtualization
+slug: adding-loader-combobox-remote-data-virtualization
+position:
+tags: combobox, ui-for-blazor, templates, nodatatemplate, header-template, loader
+res_type: kb
+ticketid: 1693304
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>ComboBox for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+When using the [ComboBox](https://www.telerik.com/blazor-ui/documentation/components/combobox/overview) component in UI for Blazor with remote data loading and virtualization, the dropdown briefly displays the "No data" message while waiting for the response. This behavior can confuse users who may assume there is no data available when it is still loading. Additionally, during virtual scrolling or filtering, the absence of a loading indicator can lead to user frustration as they cannot perceive ongoing data fetch operations.
+
+## Solution
+
+To address the missing loading indicator issue in the TelerikComboBox component, follow these steps:
+
+### 1. Display a Loading Indicator During Remote Data Fetch
+Use the `HeaderTemplate` property of the ComboBox component to show a loading indicator. Add a boolean flag to track the loading state and update it dynamically during data fetch operations. Use a CSS rule to disable the default "No Data" message.
+
+### 2. Clear Old Items During Filtering/Search
+Modify the visibility of old items using CSS rules and conditionally toggle their appearance based on the loading state.
```

Comment:
We use a numbered list for the steps:
https://www.telerik.com/blazor-ui/documentation/knowledge-base/grid-search-button-click

Headings are acceptable if you want to show two alternative ways to achieve the same result, with two separate examples:
https://www.telerik.com/blazor-ui/documentation/knowledge-base/grid-get-filtered-data#solution